src/math/BesselK.cpp

```diff
@@ -0,0 +1,1037 @@
+        //           KODE   - a parameter to indicate the scaling option
+        //                    KODE=1 returns Y(I)=       K/sub(FNU+I-1)/(X),
+        //                                        I=1,...,N
+        //                    KODE=2 returns Y(I)=EXP(X)*K/sub(FNU+I-1)/(X),
+        //                                        I=1,...,N
+        //           N      - number of members in the sequence, N .GE. 1
+        //
+        //         Output     Y is double precision
+        //           Y      - a vector whose first N components contain values
+        //                    for the sequence
+        //                    Y(I)=       k/sub(FNU+I-1)/(X), I=1,...,N  or
+        //                    Y(I)=EXP(X)*K/sub(FNU+I-1)/(X), I=1,...,N
+        //                    depending on KODE
+        //           NZ     - number of components of Y set to zero due to
+        //                    underflow with KODE=1,
+        //                    NZ=0   , normal return, computation completed
+        //                    NZ .NE. 0, first NZ components of Y set to zero
+        //                             due to underflow, Y(I)=0.0D0, I=1,...,NZ
+        //
+        //     Error Conditions
+        //         Improper input arguments - a fatal error
+        //         Overflow - a fatal error
+        //         Underflow with KODE=1 -  a non-fatal error (NZ .NE. 0)
+        //
+        // ***REFERENCES  F. W. J. Olver, Tables of Bessel Functions of Moderate
+        //                 or Large Orders, NPL Mathematical Tables 6, Her
+        //                 Majesty's Stationery Office, London, 1962.
+        //               N. M. Temme, On the numerical evaluation of the modified
+        //                 Bessel function of the third kind, Journal of
+        //                 Computational Physics 19, (1975), pp. 324-337.
+        // ***ROUTINES CALLED  D1MACH, DASYIK, DBESK0, DBESK1, DBSK0E, DBSK1E,
+        //                    DBSKNU, I1MACH, XERMSG
+        // ***REVISION HISTORY  (YYMMDD)
+        //    790201  DATE WRITTEN
+        //    890531  Changed all specific intrinsics to generic.  (WRB)
+        //    890911  Removed unnecessary intrinsics.  (WRB)
+        //    890911  REVISION DATE from Version 3.2
+        //    891214  Prologue converted to Version 4.0 format.  (BAB)
+        //    900315  CALLs to XERROR changed to CALLs to XERMSG.  (THJ)
+        //    920501  Reformatted the REFERENCES section.  (WRB)
+        //    170203  Converted to C++, and modified to only cover n==1 option. (MJ)
+        // ***END PROLOGUE  DBESK
+
+        const int nulim[2] = { 35,70 };
+
+        const double elim = -std::log(std::numeric_limits<double>::min() * 1.e3);
+        const double xlim = std::numeric_limits<double>::min() * 1.e3;
+
+        assert(fnu >= 0.);
+        assert(x > 0.);
+
+        if (x < xlim)
+            throw std::runtime_error("DBESK OVERFLOW, FNU OR N TOO LARGE OR X TOO SMALL");
```

Comment:
I'm a little confused by this error, because it only checks that x is too small, but then says that it could also be that fnu or n is too large?

Reply:
They had one error message that can happen from a few cases in the code.  I didn't bother to change the error message.  But yeah, in this case, it's just the x too small part.  Below there are other ones that can hit if nu is very large.